### PR TITLE
project clusterresourcequota into applicable namespaces

### PR DESCRIPTION
**Pull request by deads2k**
Last commit only.  Builds on https://github.com/openshift/origin/pull/9558 and https://github.com/openshift/origin/pull/9588.

Adds `appliedclusterresourcequotas` as a projection of `clusterresourcequotas` that apply 

@openshift/api-review Let's sort out name.  Other possibilities that came up: applicableclusterresourcequota (jessica didn't care for it), associatedclusterresourcequota (doesn't seem any better)

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

File: pkg/quota/registry/appliedclusterresourcequota/filter.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	namespace, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		return nil, kapierrors.NewBadRequest("namespace is required")
+	}
+
+	// TODO max resource version?  watch?
+	list := &quotaapi.ClusterResourceQuotaList{}
+	matcher := clusterresourcequotaregistry.Matcher(oapi.ListOptionsToSelectors(options))
+	quotaNames, _ := r.quotaMapper.GetClusterQuotasFor(namespace)
+	fmt.Printf("#### 10a quotaNames: %v\n", quotaNames)
```
Comment (soltysh) [inline pkg/quota/registry/appliedclusterresourcequota/filter.go]: Debug leftoever.

File: pkg/quota/api/validation/validation_test.go [hunk 2]
```diff
@@ -1,9 +1,105 @@
...
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/util/validation/field"
 	"testing"
+
+	quotaapi "github.com/openshift/origin/pkg/quota/api"
 )
 
-func TestClusterQuota(t *testing.T) {
+func TestValidationClusterQuota(t *testing.T) {
```
Comment (soltysh) [inline pkg/quota/api/validation/validation_test.go]: :+1:

File: pkg/quota/api/types.go [hunk 3]
```diff
@@ -54,6 +54,30 @@ type ClusterResourceQuotaList struct {
...
 
+// AppliedClusterResourceQuota mirrors ClusterResourceQuota at a namespace scope, for projection
+// into a namespace
+type AppliedClusterResourceQuota struct {
+	unversioned.TypeMeta
+	// Standard object's metadata.
+	kapi.ObjectMeta
+
+	// Spec defines the desired quota
+	Spec ClusterResourceQuotaSpec
```
Comment (soltysh) [inline pkg/quota/api/types.go]: I'll leave it here as a food for thought, how about just having the Status if we want to show the current state? I am aware it's easier to map entire object, how it's done right now.

**Comment by soltysh** [comment]
One debug leftover, otherwise LGTM.

File: pkg/quota/api/types.go (hunk 3 above)
Comment (liggitt) [inline pkg/quota/api/types.go]: definitely want the spec as well... the UI will use the spec to show things like "cluster quota limit of 1000 pods, this project is using 5, other projects are using 900"

Comment (soltysh) [inline pkg/quota/api/types.go]: Makes sense.

**Comment by liggitt** [comment]
other alternatives considered:

* selectingclusterresourcequotas
* relatedclusterresourcequotas
* localclusterresourcequotas

appliedclusterresourcequotas seems the best of the bunch

**Comment by deads2k** [comment]
[merge]

**Comment by smarterclayton** [comment]
Add a better description to the object godoc, including how it is used

**Comment by deads2k** [comment]
> Add a better description to the object godoc, including how it is used

done.

**Comment by smarterclayton** [comment]
Impeccable.

**Comment by deads2k** [comment]
re[test]

**Comment by deads2k** [comment]
https://github.com/openshift/origin/issues/9512 re[test]

**Comment by deads2k** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/2802/ [merge]

**Pull request merged**